directdemod/merger/merger.py

```diff
@@ -0,0 +1,256 @@
+'''
+image merger
+'''
+import numpy as np
+import matplotlib.pyplot as plt
+import cartopy
+import cartopy.crs as ccrs
+import os
+import config
+
+from datetime import datetime, timedelta
+from pyorbital.orbital import Orbital
+from PIL import Image
+from lib_checker import Checker
+from scipy import ndimage
+
+'''
+This class provides an API for merging multiple images.
+It extracts needed information and projects images on mercator
+projection.
+'''
+
+class ImageMerger:
+
+    '''
+    Class for merging multiple images
+    '''
+
+    def __init__(self, tle_file, resolution=config.RESOLUTION, aux_file_name="temp_image_file.png"):
+
+        '''Initialize the object
+
+        Args:
+            tle_file (:obj:`string`): path to the tle file
+            resolution (:obj:`int`, optional): resolution of the output image
+            aux_file_name (:obj:`string`, optional): auxiliary file name
+        '''
+
+        Checker.check_libs()
+
+        self.is_basemap = Checker.check_basemap()
+        self.is_cartopy = Checker.check_cartopy()
+        self.resolution = resolution
+        self.tle_file   = tle_file
+        self.aux_file_name = aux_file_name
+
+    def merge_images(self, objs, whole=False):
+
+        '''Merge multiple images
+
+        Args:
+            objs (:obj:`tuple(np.array, tuple(float, float), float)`): set of objects representing images
+            whole (:obj:`bool`, optinal): flag to generate whole image
+
+        Returns:
+            :obj:`PIL.Image`: merged image
+        '''
+
+        if objs is None:
+            raise ValueError("Passed objects are null.")
+
+        projection = plt.figure()
+
+        axes = plt.axes(projection=ccrs.PlateCarree())
+        axes.coastlines(resolution='50m', color=config.COLOR, linewidth=0.2)
+        axes.add_feature(cartopy.feature.BORDERS, linestyle='-', edgecolor=config.COLOR, linewidth=0.2)
+        axes.set_extent([-180, 180, -90, 90])
+
+        self.left_ex  = 180
+        self.right_ex = -180
+        self.bot_ex   = 90
+        self.top_ex   = -90
+
+        for obj in objs:
+            image  = obj[0]
+            center = obj[1]
+            degree = obj[2]
+
+            if self.is_cartopy:
+                img = ndimage.rotate(image, degree - 180, cval=255)
+                dx = img.shape[0]*4000/2*0.81 # in meters
```

Comment:
This particular set of constants (like: 0.81) worked only for one test case. So instead of following method may I recommend that you store the image info (maybe in another file along with the image?) where after NOAA has processed the image it will just write metadata like: Image center long/lat etc. Then when we are able to extract images out of other sats, same merger code could be used.

Reply:
Well, this is quite an accurate remark. And actually it will be good decision. I have already thought, that stroring all image info in it's name is not that comfortable, and even less flexible. So maybe we shall define some sort of unified data file, for example some json. It will store all information on the image and then it this file would be used in other applications, like merger. So they won't depend on any image name and so on.

Reply:
I have resolved the easy ones and will take on issue of image storing.

Reply:
I have stored all data on image in separate json file. Now data does not depend on algorithms implementation. Merger could be used on any image which has such descriptor file.
I have also fixed the `Experiment 8` example.

In addition merger does depend only on certain fields in descriptor files, so the file itself can be extended, other fields could be added and so on. Wanted also to note, that field names maybe should be changed, because I've chosen the first I've thought about.